src/Module.ts

```diff
@@ -45,9 +45,10 @@ export interface CommentDescription { block: boolean, text: string, start: numbe
 export interface ExportDescription { localName: string, identifier?: string }
 export interface ReexportDescription { localName: string, start: number, source: string, module: Module }
 
-function tryParse (module: Module, acornOptions: Object) {
+function tryParse (module: Module, acornOptions: Object, acornInjects: Function[]) {
```

Comment:
These injections don't need to happen for every module parse - rather the `acornInjects.reduce(acc, plugin) => plugin(acc))` step can be done in the Graph constructor I think rather?

Reply:
Acorn plugins are not composable in that they expect the `acorn` object to be passed as an argument ([example](https://github.com/victor-homyakov/acorn-object-rest-spread/blob/a80f50b7a91360d168bf9cc977eaedd4cc84d973/inject.js)). I think I'll need to import `acorn` in `Graph.js`, produce the final `acorn` object there, and then pass it to `tryParse`. Sound good?

Reply:
Sounds sensible! Alternatively because we have live bindings in ES Modules, it could be possible to have Graph.ts explicitly export its own acorn variable which is imported by Module.ts.

Reply:
Actually that said, much better to have class-level encapsulation actually here.

Reply:
I wasn't able to figure out how to pass the entire `acorn` object to `tryParse`. I think it's related to the fact that it's declared as a namespace in [@types/acorn](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/acorn/index.d.ts#L10), but I don't know TypeScript well enough to know for sure.

I ended up passing the `parse` function instead.